### PR TITLE
feat: add db2 data modules with sample

### PR DESCRIPTION
**Pull request by tisla1080**
- Added base db2 data module for IBM DB2 Instances
- Users just need to provide database credentials in property files along with entity and any specific repository methods
- Added sample usage data model 
- Tested on local using local db2 instance so connection is working
- Add ReadMe

### CODE REVIEW

File: data/data-samples/sample-data-db2-book/src/main/resources/data-book-application.properties [hunk 1]
```diff
@@ -0,0 +1,6 @@
+spring.datasource.url=some-url
+spring.datasource.username=some-username
+spring.datasource.password=some-password
+spring.datasource.driver-class-name=com.ibm.db2.jcc.DB2Driver
+spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.DB2Dialect
+spring.jpa.hibernate.ddl-auto=none
```
Comment (andCastStatic) [inline data/data-samples/sample-data-db2-book/src/main/resources/data-book-application.properties]: add a new line at the end of the file.

File: data/synapse-data-db2/pom.xml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            <plugin>
+                <artifactId>maven-checkstyle-plugin</artifactId>
+            </plugin>
+            <plugin>
+                <artifactId>maven-pmd-plugin</artifactId>
+            </plugin>
+        </plugins>
+    </reporting>
+
+</project>
```
Comment (andCastStatic) [inline data/synapse-data-db2/pom.xml]: add a new line at the end of the file

File: data/synapse-data-db2/src/main/java/io/americanexpress/synapse/data/db2/config/BaseDb2Config.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+ */
+@Configuration
+@EnableTransactionManagement
+public abstract class BaseDb2Config {
+
+    /**
+     * Contains environment variables needed for connection.
+     */
+    @Autowired
+    protected Environment environment;
```
Comment (andCastStatic) [inline data/synapse-data-db2/src/main/java/io/americanexpress/synapse/data/db2/config/BaseDb2Config.java]: is it possible to do constructor autowiring?

private final Environment environment;

public BaseDb2Condif(Environment environment) { this.environment = environment; }

File: data/data-samples/sample-data-db2-book/pom.xml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+            <groupId>io.americanexpress.synapse</groupId>
+            <artifactId>synapse-data-db2</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-test</artifactId>
+        </dependency>
+    </dependencies>
+
+</project>
```
Comment (andCastStatic) [inline data/data-samples/sample-data-db2-book/pom.xml]: add new line at the end of the file

File: data/synapse-data-db2/src/main/java/io/americanexpress/synapse/data/db2/config/BaseDb2Config.java (hunk 3 above)
Comment (tisla1080) [reply data/synapse-data-db2/src/main/java/io/americanexpress/synapse/data/db2/config/BaseDb2Config.java]: done

File: data/data-samples/sample-data-db2-book/pom.xml (hunk 4 above)
Comment (tisla1080) [reply data/data-samples/sample-data-db2-book/pom.xml]: done

File: data/synapse-data-db2/pom.xml (hunk 2 above)
Comment (tisla1080) [reply data/synapse-data-db2/pom.xml]: done

File: data/data-samples/sample-data-db2-book/src/main/resources/data-book-application.properties (hunk 1 above)
Comment (tisla1080) [reply data/data-samples/sample-data-db2-book/src/main/resources/data-book-application.properties]: done

File: data/data-samples/sample-data-db2-book/src/main/resources/data-book-application.properties [hunk 5]
```diff
@@ -0,0 +1,6 @@
+spring.datasource.url=some-url
```
Comment (guttff) [inline data/data-samples/sample-data-db2-book/src/main/resources/data-book-application.properties:13]: copyright

File: pom.xml [hunk 6]
```diff
@@ -967,6 +972,23 @@
...
+            </dependency>
+            <dependency>
+                <groupId>org.springframework.boot</groupId>
+                <artifactId>spring-boot-starter-data-jpa</artifactId>
+                <version>${spring-boot-framework.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>com.ibm.db2</groupId>
+                <artifactId>jcc</artifactId>
+                <version>11.5.0.0</version>
```
Comment (guttff) [inline pom.xml]: move version to the parent pom

File: data/data-samples/sample-data-db2-book/src/main/java/io/americanexpress/data/book/config/BookDataConfig.java [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+import org.springframework.core.env.Environment;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+import org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean;
+
+/**
+ * {@code BookDataConfig} is the configuration class to load all the properties for the book data module.
+ */
+@Configuration
+@PropertySource("classpath:/data-book-application.properties")
+@EnableJpaRepositories(basePackages = {"io.americanexpress.data.book"})
```
Comment (guttff) [inline data/data-samples/sample-data-db2-book/src/main/java/io/americanexpress/data/book/config/BookDataConfig.java]: alphabetize

Comment (guttff) [inline data/data-samples/sample-data-db2-book/src/main/java/io/americanexpress/data/book/config/BookDataConfig.java]: if only one class then no need for the curly braces

File: data/data-samples/sample-data-db2-book/src/main/java/io/americanexpress/data/book/config/BookDataConfig.java [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    /**
+     * Set the packages to Scan property to the entityManagerFactory.
+     *
+     * @param entityManagerFactory the entity manager factory
+     */
+    @Override
+    protected void setPackagesToScan(LocalContainerEntityManagerFactoryBean entityManagerFactory) {
+        entityManagerFactory.setPackagesToScan("io.americanexpress.data.book");
```
Comment (guttff) [inline data/data-samples/sample-data-db2-book/src/main/java/io/americanexpress/data/book/config/BookDataConfig.java]: should be a constant if used more than once

File: data/data-samples/sample-data-db2-book/src/main/resources/data-book-application.properties (hunk 5 above)
Comment (tisla1080) [reply data/data-samples/sample-data-db2-book/src/main/resources/data-book-application.properties:13]: done

File: pom.xml (hunk 6 above)
Comment (tisla1080) [reply pom.xml]: done

File: data/data-samples/sample-data-db2-book/src/main/java/io/americanexpress/data/book/config/BookDataConfig.java (hunk 7 above)
Comment (tisla1080) [reply data/data-samples/sample-data-db2-book/src/main/java/io/americanexpress/data/book/config/BookDataConfig.java]: done

Comment (tisla1080) [reply data/data-samples/sample-data-db2-book/src/main/java/io/americanexpress/data/book/config/BookDataConfig.java]: done

File: data/data-samples/sample-data-db2-book/src/main/java/io/americanexpress/data/book/config/BookDataConfig.java (hunk 8 above)
Comment (tisla1080) [reply data/data-samples/sample-data-db2-book/src/main/java/io/americanexpress/data/book/config/BookDataConfig.java]: done

**Review by guttff — APPROVED** [review]

**Pull request merged**